src/quickfix.c

```diff
@@ -36,6 +36,7 @@ struct qfline_S
     int		qf_end_col;	// column when the error has range or zero
     int		qf_nr;		// error number
     char_u	*qf_module;	// module name for this error
+    char_u	*qf_fname;	// filename for a line
```

Comment:
Creating a copy of the filename for each quickfix entry will significantly increase the memory used for the quickfix list.  Is this really needed?

Reply:
Thanks for the comment. My bad that I wasn't aware of the memory usage at the time. My thought was that the list is from LSP response. If the filenames already occupy the memory before `setqflist`, then it should be ok to stay there(copy & release the original one by caller) and hold by qflist.

Another approach would be doing a comparison with `buf->b_ffname` and only copy the filename if they are different, or add an option on `setqflist` to tell if caller want it to do so. This still leads to another doubt that if the time of comparison on each entry really worth.